### ISSUE TITLE
Separate measurement and interpretation

### ISSUE THREAD

**Issue by bassosimone** [ooni/probe-engine#13]
From https://github.com/measurement-kit/measurement-kit/issues/1712:

> See https://github.com/measurement-kit/measurement-kit/issues/1704#issuecomment-447845953
>
> See also this message of mine on Slack, which elicited lots of positive feedback:
>
>> 6. we can refrain from changing the code that decides whether something is blocked for now, but we should change that based on mining data in the future. I do not like that our data format does not clearly distinguish between (a) this is what I did measure and (b) this is what I _believe_ what I measured means. This is unfortunate. *Our data format should clearly distinguish between facts, i.e. what was measured, and inferences based on such facts*.

**Comment by bassosimone** [comment]
This seems to me deeply related to #12

**Comment by bassosimone** [comment]
Also, from https://github.com/measurement-kit/measurement-kit/issues/1161#issuecomment-449313430:
 
>> 7. it is possible to argue that repeating a measurement is the right thing to do when, say, the network is bad and that repeating a failed measurement could have provided the user with the right result. I disagree with that, and I believe that the right result does not matter much, if the methodology is wrong. What we should actually do is (a) measure and (b) reason on the measurement. In my view, these two pieces of codes should probably be quite decoupled. *Repeating a measurement is not bad, provided that we not discard the information that the measurement failed N times, because that is also valuable data*. 

Therefore, one should be able to see from the results that we did retry the measurement.

**Comment by bassosimone** [comment]
We have explained how this could be done in #92. We still need to work onto re-running measurements, tho, as this has not been addressed yet.

**Comment by bassosimone** [comment]
This can also be considered one of the guiding principles of #87. To improve on device data analysis we needed to separate measurements (github.com/ooni/netx) and interpretation (github.com/ooni/probe-engine). While it is true that we will keep a backwards compatible data format, we've made enough progress towards this goal that we can close this rather abstract issue with confidence that the separation is as robust as two different repositories.
